phantoms/MR_XCAT_qMRI/sim_ivim_sig.py

```diff
@@ -364,76 +366,125 @@ def XCAT_to_MR_DCE(XCAT, TR, TE, bvalue, D, f, Ds, b0=3, ivim_cont = True):
-            'f': np.mean(fim[selector], axis=0),
-            'Dp': np.mean(Dpim[selector], axis=0),
-            'data': signals
-        }
-    generic_data['config'] = {
-        'bvalues': bvalue.tolist()
-    }
-    with open('generic.json', 'w') as f:
-        json.dump(generic_data, f, indent=4)
+    parser = argparse.ArgumentParser(description=f"""
+    A commandline for generating a 4D IVIM phantom as nifti file
+                                    """)
+
+    def parse_bvalues_file(file_path):
+        """Used for passing the JSON file"""
+        if not os.path.exists(file_path):
+            raise argparse.ArgumentTypeError(f"File '{file_path}' does not exist")
+
+        try:
+            with open(file_path, "r") as file:
+                bvalues_list = json.load(file)
+                if not isinstance(bvalues_list, list):
+                    raise argparse.ArgumentTypeError("JSON file does not contain a list of b-values")
+                for bvalue in bvalues_list:
+                    if not isinstance(bvalue, list):
+                        raise argparse.ArgumentTypeError("bvalues in JSON file are not lists")
+                    for v in bvalue:
+                        if not isinstance(v, float):
+                            raise argparse.ArgumentTypeError("Values in lists are not float")
+        except json.JSONDecodeError as e:
+            raise argparse.ArgumentTypeError(f"Invalid JSON file: {e}")
+
+        return bvalues_list
 
 
-    nifti_img = nib.Nifti1Image(sig, affine=res)  # Replace affine if necessary
-    # Save the NIfTI image to a file
-    nifti_img.header.set_data_dtype(np.float64)
-    if not motion:
-        output_file = 'output.nii.gz'  # Replace with your desired output file name
-    elif interleaved:
-        output_file = 'output_resp_int.nii.gz'  # Replace with your desired output file name
+    parser.add_argument("-b", "--bvalue", type=float,
+                        nargs="+",
+                        help="B values (list of of numbers)")
+    parser.add_argument("-bf", "--bvalues-file", metavar="FILE", type=parse_bvalues_file,
+                    help='JSON file containing the b-values')
+    parser.add_argument("-n", "--noise", type=float, default=0.0005, help="Noise")
+    parser.add_argument("-m", "--motion", action="store_true", help="Motion flag")
+    parser.add_argument("-i", "--interleaved", action="store_true", help="Interleaved flag")
+    args = parser.parse_args()
+
+    if args.bvalues_file and args.bvalue:
+        raise argparse.ArgumentError(None, "Arguments --bvalues-file and --bvalues are mutually exclusive")
+    
+    bvalues = None
+    if args.bvalues_file:
+        bvalues = args.bvalues_file
+    elif args.bvalue:
+        bvalues = [args.bvalue]
     else:
-        output_file = 'output_resp.nii.gz'  # Replace with your desired output file name
+        bvalues = [[0., 1, 2, 5, 10, 20, 30, 50, 75, 100, 150, 250, 350, 400, 550, 700, 850, 1000]]
```

Comment:
It seems odd to have a double array, why is this?

Reply:
I had to make it a double array so that I can have one implementation with when the value is gotten from the files. Check line 420.

Reply:
Oh I see. I guess it'll change with an object implementation anyway.